tests/datasets/test_cropharvest.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import builtins
+import os
+import shutil
+from pathlib import Path
+from typing import Any
+
+import matplotlib.pyplot as plt
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from pytest import MonkeyPatch
+
+import torchgeo.datasets.utils
+from torchgeo.datasets import CropHarvest, DatasetNotFoundError
+
+pytest.importorskip("h5py", minversion="3")
+
+
+def download_url(url: str, root: str, filename: str, md5: str) -> None:
+    shutil.copy(url, os.path.join(root, filename))
+
+
+class TestCropHarvest:
+    file_dict = {
```

Comment:
Instead of replacing the entire dict, you could replace just the keys you need to replace (url, md5) using `monkeypatch.setitem`